src/tools/nbench/nbench.cpp

```diff
@@ -157,8 +172,34 @@ OPTIONS
             cout << op_info.first << ": " << op_info.second << " ops" << endl;
         }
     }
+    else if (!directory.empty())
+    {
+        vector<string> models;
+        file_util::iterate_files(directory,
+                                 [&](const string& file, bool is_dir) {
+                                     if (!is_dir)
+                                     {
+                                         models.push_back(file);
+                                     }
+                                 },
+                                 true);
+        for (const string& m : models)
+        {
+            try
+            {
+                shared_ptr<Function> f = deserialize(m);
+                cout << "Benchmarking " << m << ", " << backend << " backend, " << iterations
+                     << " iterations.\n";
+                run_benchmark(f, backend, iterations, timing_detail);
+            }
+            catch (...)
```

Comment:
If one of the models throws an exception in the benchmark, we should let it emit rather than ignoring it with this catch.

Reply:
Do we want to continue on? I think we should. We can catch, emit, and continue.

Reply:
The plan is to aggregate everything into tables in a web page generated by the test.

Reply:
Sure, continuing on is fine. The catch can update the GUI/std::err/out with a notice of which model failed.

Reply:
👍 , this one is good to go.